src/c++/vdo/base/dm-vdo-target.c

```diff
@@ -1086,25 +1106,8 @@ static void vdo_io_hints(struct dm_target *ti, struct queue_limits *limits)
 	 * The value is used by dm-thin to determine whether to pass down discards. The block layer
 	 * splits large discards on this boundary when this is set.
 	 */
-#ifndef VDO_UPSTREAM
-#undef VDO_USE_ALTERNATE
-#if defined(RHEL_RELEASE_CODE) && defined(RHEL_MINOR) && (RHEL_MINOR < 50)
-#if (RHEL_RELEASE_CODE < RHEL_RELEASE_VERSION(10, 0))
-#define VDO_USE_ALTERNATE
-#endif
-#else /* !RHEL_RELEASE_CODE */
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 10, 0))
-#define VDO_USE_ALTERNATE
-#endif
-#endif /* !RHEL_RELEASE_CODE */
-#endif /* !VDO_UPSTREAM */
-#ifdef VDO_USE_ALTERNATE
-	limits->max_discard_sectors =
-		(vdo->device_config->max_discard_blocks * VDO_SECTORS_PER_BLOCK);
-#else
```

Comment:
Removing this is fine and good, but you should mention it in the commit message.

Reply:
Unfortunately I figured out a reason to keep this VDO_USE_ALTERNATE block. Although all on pfarms are 6.10 or higher, so might have some older F39 and F40 machines (like my build machine) which are not as caught up on the kernels. I don't really want to have to constantly update my working machines, so I'd prefer the VDO_USE_ALTERNATE block for 6.10 remains.

We can ask the group on Monday for other opinions, too.